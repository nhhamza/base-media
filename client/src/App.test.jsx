import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import { Provider } from 'react-redux';

import configureStore from 'redux-mock-store';

import AppConnected from './App';
import { mount } from 'enzyme';

const mockStore = configureStore([]);

jest.mock('react-qr-reader');

describe('App component', () => {
  const generateElement = (store, route) => {
    return mount(
      <Provider store={store}>
        <MemoryRouter initialEntries={[`${route}`]}>
          <AppConnected />
        </MemoryRouter>
      </Provider>
    );
  };

  describe('Authenticated', () => {
    let MOCK_STORE;

    const getMockStoreFromUser = () =>
      mockStore({
        callCenter: {
          callCenterList: [
            { id: '1', name: 'operator1', number: '1111131313' },
          ],
          error: null,
        },
      });

    beforeEach(() => {
      MOCK_STORE = getMockStoreFromUser({
        error: null,
      });
    });

    describe('/no-real-route route', () => {
      it('renders properly', () => {
        expect(generateElement(MOCK_STORE, '/no-real-route')).toMatchSnapshot();
      });
    });

    describe('/ route', () => {
      it('renders properly', () => {
        expect(generateElement(MOCK_STORE, '/')).toMatchSnapshot();
      });
    });

    /*describe('/help route', () => {
      it('renders properly', () => {
        expect(generateElement(MOCK_STORE, '/help')).toMatchSnapshot();
      });
    });*/
  });
});
