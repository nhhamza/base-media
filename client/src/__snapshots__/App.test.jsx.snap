// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component Authenticated / route renders properly 1`] = `ReactWrapper {}`;

exports[`App component Authenticated /no-real-route route renders properly 1`] = `ReactWrapper {}`;
